src/app-exception-filters/main-exception.filter.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+
+@Catch()
+export class MainExceptionFilter implements ExceptionFilter {
+  private readonly logger = new Logger();
+
+  catch(exception: Error, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    const status =
+      exception instanceof HttpException ? exception.getStatus() : 500;
+
+    this.logger.error(exception.stack, MainExceptionFilter.name);
```

Comment:
jak wrzucisz po prostu sam error to nie zadziala przypadkiem tak samo?

Reply:
z tego co sie oreitnuje to jak przekarzesz caly error to on sam sobie stack wyciagnie

Reply:
nie wyciąga :) przetestowałem i rzuciłem cały exception to pobrało tylko nazwe błędu i message
Aby stack był załączony to muszę rzucić stack

Reply:
TAK! errory i stack w JS są mega słabe. Dlatego zmiana w ES2022 i to pole... Ten cały "stack" nic nie mowi. A jeszce z tego co widziałem na poprzednim projekcie, bez totania error.name to Nest nie pokazuje nawet customowej nazwy błędu (klasy)...